crates/trie/sparse/src/trie.rs

```diff
@@ -413,146 +414,12 @@ impl SparseTrieInterface for SerialSparseTrie {
         self.nodes.reserve(additional);
     }
 
-    fn reveal_node(
-        &mut self,
-        path: Nibbles,
-        node: TrieNode,
-        masks: TrieMasks,
-    ) -> SparseTrieResult<()> {
-        // If the node is already revealed and it's not a hash node, do nothing.
-        if self.nodes.get(&path).is_some_and(|node| !node.is_hash()) {
-            return Ok(())
-        }
-
-        if let Some(tree_mask) = masks.tree_mask {
-            self.branch_node_tree_masks.insert(path, tree_mask);
+    fn reveal_nodes(&mut self, mut nodes: Vec<RevealedSparseNode>) -> SparseTrieResult<()> {
+        nodes.sort_unstable_by(|a, b| a.path.cmp(&b.path));
+        self.nodes.reserve(nodes.len());
```

Comment:
During multiproof revealing we reserve nodes then call reveal_nodes, so not sure if we should include this

Reply:
True, it's not necessary here, will remove it :+1: